packages/zoe/test/unitTests/contractSupport/test-prepare-revocable.js

```diff
@@ -0,0 +1,124 @@
+// Modeled on test-heap-classes.js
+
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+
+// eslint-disable-next-line import/order
+import { M } from '@endo/patterns';
+import { makeHeapZone } from '@agoric/base-zone/heap.js';
+import { prepareRevocableKit } from '../../../src/contractSupport/prepare-revocable.js';
+
+const UpCounterI = M.interface('UpCounter', {
+  incr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+const DownCounterI = M.interface('DownCounter', {
+  decr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+test('test revoke defineVirtualExoClass', t => {
+  const zone = makeHeapZone('rootZone');
+
+  const makeUnderlyingUpCounter = zone.exoClass(
+    'UpCounter',
+    UpCounterI,
+    /** @param {number} [x] */
+    (x = 0) => ({ x }),
+    {
+      incr(y = 1) {
+        const { state } = this;
+        state.x += y;
+        return state.x;
+      },
+    },
+  );
+
+  const makeRevocableUpCounterKit = prepareRevocableKit(
+    zone,
+    'UpCounter',
+    'UpCounter',
```

Comment:
multiple arguments of the same type next to each other means there's little to no help from the machine in getting them in the right order.

Having 2 un-named optional args at the end also suggests that more may come later. And it means that to specify the last one, you have to put `undefined` in the 2nd-to-last spot.

Maybe put all the args but `zone` in an options object?

Reply:
I moved only the optional parameters into the options object, which is my normal habit for options objects. If there are too many remaining required parameters, I might still try to make these non-positional somehow. But in this case, there were few enough required parameters left that I kept them positional.

PTAL. I'm resolving under the theory that this is done enough. If you disagree please unresolve and comment. Thanks.